upload_vehicle_data.py

```diff
@@ -0,0 +1,327 @@
+            get_remote_file_size(remote_user, remote_ip, rosbag, password)
+            for rosbag in rosbag_list
+        )
+        estimated_time = get_estimated_upload_time(
+            total_size_gb, bandwidth_mbps
+        )
+
+        print(
+            f'Found {len(rosbag_list)} files to upload with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        print(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+        print(f'Estimated upload time: {estimated_time:.2f} hours.')
+        confirm = input('Do you want to proceed to upload? (yes/no): ')
+
+        if confirm.lower() != 'yes':
+            print('Upload aborted.')
+            return
+
+        logging.info(
+            f'Starting upload of {len(rosbag_list)} files with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        logging.info(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+
+        for rosbag in rosbag_list:
+            success = compress_and_transfer_rosbag(
+                remote_user,
+                remote_ip,
+                rosbag,
+                remote_temp_directory,
+                vdi_upload_directory,
+                password,
+            )
+            if not success:
+                logging.error(
+                    f'Failed to process {rosbag}. Continuing with next file.'
+                )
+            else:
+                logging.info(f'Successfully processed {rosbag}.')
+
+    finally:
+        # Delete the remote temporary directory
+        delete_remote_directory(
+            remote_user, remote_ip, remote_temp_directory, password
+        )
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Automate the compression and upload of rosbags.'
+    )
+    parser.add_argument('remote_user', help='Username for the remote machine')
+    parser.add_argument(
+        'password', help='Password for SSH and rsync operations'
+    )
+    parser.add_argument(
+        '-remote_temp_directory',
+        default='/mnt/mydrive/rosbags/temp',
+        help='Remote temporary directory for storing compressed files',
+    )
+    parser.add_argument(
+        '-iperf_server_ip',
+        default='129.215.117.104',
+        help='IP address of the iperf3 server for bandwidth measurement',
+    )
+    parser.add_argument(
+        '-remote_ip',
+        default='129.215.117.104',
+        help='IP address of the remote machine (default: 129.215.117.104)',
+    )
+    parser.add_argument(
+        '-remote_directory',
+        default='/mnt/mydrive/rosbags',
+        help=(
+            'Directory on the remote machine containing rosbags '
+            '(default: /mnt/mydrive/rosbags)'
+        ),
+    )
+    parser.add_argument(
+        '-vdi_upload_directory',
+        default='/mnt/vdb/data',
+        help=(
+            'VDI directory for uploading compressed files '
+            '(default: /mnt/vdb/data)'
+        ),
+    )
+
+    args = parser.parse_args()

```

Comment:
Lets create a YAML file where these parameters are defined and parse this file instead. This way we can have  multiple files to define different remote machines, such as `vehicle_data_params.yaml`  or `rhaegal_data_params.yaml` for debugging.

Reply:
change the argument parsing to YAML style